source/extensions/filters/common/ratelimit_config/ratelimit_config.cc

```diff
@@ -111,6 +138,20 @@ void RateLimitPolicy::populateDescriptors(const Http::RequestHeaderMap& headers,
       descriptor.entries_.emplace_back(std::move(entry));
     }
   }
+
+  // Populate hits_addend if set.
+  if (hits_addend_provider_ != nullptr) {
+    const ProtobufWkt::Value hits_addend =
+        hits_addend_provider_->formatValueWithContext({&headers}, stream_info);
+    if (hits_addend.has_number_value()) {
+      descriptor.hits_addend_ = static_cast<uint32_t>(hits_addend.number_value());
+    } else {
+      ENVOY_LOG(warn, "hits_addend must be a number");
```

Comment:
This can heavily log spam if there is a config mistake. This should be a warn every of some type. Optimally we would catch as much as possible at config load time which isn't happening.

Reply:
> This can heavily log spam if there is a config mistake. This should be a warn every of some type. Optimally we would catch as much as possible at config load time which isn't happening.

Currently we have no way to check the provider's expected return type. I can change the level to debug and add more clear descripription on the API.